optuna/pruners/_wilcoxon.py

```diff
@@ -167,6 +169,21 @@ def prune(self, study: "optuna.study.Study", trial: FrozenTrial) -> bool:
 
         alt = "less" if study.direction == StudyDirection.MAXIMIZE else "greater"
 
+        if alt == "less":
+            average_is_best = best_trial.value <= sum(step_values) / len(step_values)
+        else:
+            average_is_best = best_trial.value >= sum(step_values) / len(step_values)
+
         # We use zsplit to avoid the problem when all values are zero.
         p = ss.wilcoxon(diff_values, alternative=alt, zero_method="zsplit").pvalue
+
+        if p < self._p_threshold and average_is_best:
+            # WilcoxonPruner found the current trial should be pruned,
+            # but the value of the best trial was not better than
+            # the average of the current trial's intermediate values.
+            # For safety, WilcoxonPruner concludes it should not be pruned.
+            # If this if-statement is frequently satisfied,
```

Comment:
Although this is a simple question, is there a reasonable way that users can detect `this if-statement is frequently satisfied`?